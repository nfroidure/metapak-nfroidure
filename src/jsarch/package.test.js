'use strict';

const packageTransformer = require('./package');

describe('Package transformer for jsdocs', () => {
  it('should work with an empty package.json', () => {
    expect(
      packageTransformer({
        metapak: {
          data: {
            files: 'yolo.js',
          },
        },
        greenkeeper: {
          ignore: [],
        },
      })
    ).toMatchSnapshot();
  });

  it('should work with child package', () => {
    expect(
      packageTransformer({
        metapak: {
          data: {
            childPackage: true,
            files: 'yolo.js',
          },
        },
        greenkeeper: {
          ignore: [],
        },
      })
    ).toMatchSnapshot();
  });
});
