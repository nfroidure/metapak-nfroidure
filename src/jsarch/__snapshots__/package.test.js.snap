// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Package transformer for jsdocs should work with an empty package.json 1`] = `
Object {
  "devDependencies": Object {
    "jsarch": "^5.0.0",
  },
  "greenkeeper": Object {
    "ignore": Array [
      "jsarch",
    ],
  },
  "metapak": Object {
    "data": Object {
      "files": "yolo.js",
    },
  },
  "scripts": Object {
    "architecture": "jsarch yolo.js > ARCHITECTURE.md && git add ARCHITECTURE.md",
    "precz": "npm run architecture",
    "preversion": "npm run architecture",
  },
}
`;

exports[`Package transformer for jsdocs should work with child package 1`] = `
Object {
  "devDependencies": Object {
    "jsarch": "^5.0.0",
  },
  "greenkeeper": Object {
    "ignore": Array [],
  },
  "metapak": Object {
    "data": Object {
      "childPackage": true,
      "files": "yolo.js",
    },
  },
  "scripts": Object {
    "architecture": "jsarch yolo.js > ARCHITECTURE.md && git add ARCHITECTURE.md",
  },
}
`;

exports[`Package transformer for jsdocs should work with typescrit configs 1`] = `
Object {
  "devDependencies": Object {
    "jsarch": "^5.0.0",
  },
  "greenkeeper": Object {
    "ignore": Array [],
  },
  "jsarch": Object {
    "parserOptions": Object {
      "plugins": Array [
        "typescript",
      ],
    },
  },
  "metapak": Object {
    "configs": Array [
      "typescript",
    ],
    "data": Object {
      "childPackage": true,
      "files": "yolo.js",
    },
  },
  "scripts": Object {
    "architecture": "jsarch yolo.js > ARCHITECTURE.md && git add ARCHITECTURE.md",
  },
}
`;
